src/Content/Widget/CalendarExport.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * @file src/Content/Widget/CalendarExport.php
+ */
+
+namespace Friendica\Content\Widget;
+
+use Friendica\Core\L10n;
+
+require_once 'boot.php';
+require_once 'include/text.php';
+
+/**
+ * TagCloud widget
+ *
+ * @author Rabuzarus
+ */
+class CalendarExport
+{
+	/**
+	 * @brief Get the events widget.
+	 *
+	 * @return string Formated HTML of the calendar widget.
+	 */
+	public static function getHTML() {
+		$a = get_app();
+
+//		$owner_uid = $a->data['user']['uid'];
+//		// The permission testing is a little bit tricky because we have to respect many cases.
+//
+//		// It's not the private events page (we don't get the $owner_uid for /events).
+//		if (! local_user() && ! $owner_uid) {
+//			return;
+//		}
+//
+//		/*
+//		 * Cal logged in user (test permission at foreign profile page).
+//		 * If the $owner uid is available we know it is part of one of the profile pages (like /cal).
+//		 * So we have to test if if it's the own profile page of the logged in user
+//		 * or a foreign one. For foreign profile pages we need to check if the feature
+//		 * for exporting the cal is enabled (otherwise the widget would appear for logged in users
+//		 * on foreigen profile pages even if the widget is disabled).
+//		 */
+//		if (intval($owner_uid) && local_user() !== $owner_uid && ! Feature::isEnabled($owner_uid, "export_calendar")) {
+//			return;
+//		}
+//
+//		/*
+//		 * If it's a kind of profile page (intval($owner_uid)) return if the user not logged in and
+//		 * export feature isn't enabled.
+//		 */
+//		if (intval($owner_uid) && ! local_user() && ! Feature::isEnabled($owner_uid, "export_calendar")) {
+//			return;
+//		}
+		/*
+		 * All the legacy checks above seem to be equivalent to the check below, see https://ethercalc.org/z6ehv1tut9cm
+		 * If there is a mistake in the spreadsheet, please notify @MrPetovan on GitHub or by email mrpetovan@gmail.com
+		 */
+		if (!local_user()) {
```

Comment:
This would mean that visitors will not be able to export the events in the `/ cal 'module, but they should be able to export if the profile owner has enabled the' export_calendar 'option

Reply:
This is what the three permissions checks amount to after careful study, see comment above. It probably isn't intended, but it's the only logical outcome from the code.

Reply:
You have a mistake in your spreadsheet in line 3 (TRUE, FALSE, FALSE, TRUE). The third check passes because the Feature is enabled

Reply:
Ah, thanks!